quiche make check build failure

I added a quiche branch build job for 10-Dev. It looks like `make check` fails compilation:
https://ci.trafficserver.apache.org/view/10-Dev/job/10-Dev/job/quiche/2/

```
In file included from ./test/test_QPACK.cc:31:
./../../iocore/net/quic/Mock.h:656:3: error: 'virtual QUICRTTProvider* MockQUICContext::rtt_provider() const' marked 'override', but does not override
  656 |   rtt_provider() const override
      |   ^~~~~~~~~~~~
./../../iocore/net/quic/Mock.h:662:3: error: 'virtual QUICPacketProtectionKeyInfo* MockQUICContext::key_info() const' marked 'override', but does not override
  662 |   key_info() const override
      |   ^~~~~~~~
./../../iocore/net/quic/Mock.h:668:3: error: 'virtual QUICLDConfig& MockQUICContext::ld_config() const' marked 'override', but does not override
  668 |   ld_config() const override
      |   ^~~~~~~~~
./../../iocore/net/quic/Mock.h:674:3: error: 'virtual QUICCCConfig& MockQUICContext::cc_config() const' marked 'override', but does not override
  674 |   cc_config() const override
      |   ^~~~~~~~~
./../../iocore/net/quic/Mock.h:680:3: error: 'virtual QUICPathManager* MockQUICContext::path_manager() const' marked 'override', but does not override
  680 |   path_manager() const override
      |   ^~~~~~~~~~~~
make[3]: *** [Makefile:1143: test/test_qpack-test_QPACK.o] Error 1

```

## Steps for Reproduction

* docker image: ci.trafficserver.apache.org/ats/rockylinux:8 
* Branch: 10-Dev
* configure: `/configure --enable-experimental-plugins --enable-example-plugins --with-quiche=/opt/quiche --prefix=/tmp/ats --enable-werror --enable-debug`
* `make -j4 && make install && make -j4 check`

--- a/iocore/net/quic/Mock.h
+++ b/iocore/net/quic/Mock.h
@@ -255,8 +255,7 @@ public:
   MockQUICStreamManager(QUICContext *context) : QUICStreamManagerImpl(context, nullptr) {}
 
   // Override
-#if HAVE_QUICHE_H
-#else
+#ifndef HAVE_QUICHE_H
   virtual QUICConnectionErrorUPtr
   handle_frame(QUICEncryptionLevel level, const QUICFrame &f) override
   {
@@ -664,8 +663,7 @@ public:
   {
     return _config;
   }
-#if HAVE_QUICHE_H
-#else
+#ifndef HAVE_QUICHE_H
   virtual QUICRTTProvider *
   rtt_provider() const override
   {

--- a/iocore/net/quic/Mock.h
+++ b/iocore/net/quic/Mock.h
@@ -27,7 +27,11 @@
 
 #include "QUICApplication.h"
 #include "QUICStreamManager.h"
+#if HAVE_QUICHE_H
+#include "QUICStreamManager_quiche.h"
+#else
 #include "QUICStreamManager_native.h"
+#endif
 #include "QUICLossDetector.h"
 #include "QUICPacketProtectionKeyInfo.h"
 #include "QUICPinger.h"
@@ -39,6 +43,11 @@
 #include "QUICPadder.h"
 #include "QUICHandshakeProtocol.h"
 #include "QUICStreamAdapter.h"
+#if HAVE_QUICHE_H
+#include "QUICStream_quiche.h"
+#else
+#include "QUICStream_native.h"
+#endif
 
 class MockQUICContext;
 
@@ -246,6 +255,8 @@ public:
   MockQUICStreamManager(QUICContext *context) : QUICStreamManagerImpl(context, nullptr) {}
 
   // Override
+#if HAVE_QUICHE_H
+#else
   virtual QUICConnectionErrorUPtr
   handle_frame(QUICEncryptionLevel level, const QUICFrame &f) override
   {
@@ -254,6 +265,7 @@ public:
 
     return nullptr;
   }
+#endif
 
   // for Test
   int
@@ -652,6 +664,8 @@ public:
   {
     return _config;
   }
+#if HAVE_QUICHE_H
+#else
   virtual QUICRTTProvider *
   rtt_provider() const override
   {
@@ -681,6 +695,7 @@ public:
   {
     return _path_manager.get();
   }
+#endif
 
 private:
   QUICConfig::scoped_config _config;

--- a/proxy/http3/test/test_QPACK.cc
+++ b/proxy/http3/test/test_QPACK.cc
@@ -65,11 +65,19 @@ private:
 };
 
 // TODO: QUICUnidirectionalStream should be used if there
+#if HAVE_QUICHE_H
+class TestQUICStream : public QUICStreamImpl
+#else
 class TestQUICStream : public QUICBidirectionalStream
+#endif
 {
 public:
   TestQUICStream(QUICStreamId sid)
+#if HAVE_QUICHE_H
+    : QUICStreamImpl(new MockQUICConnectionInfoProvider(), sid)
+#else
     : QUICBidirectionalStream(new MockQUICRTTProvider(), new MockQUICConnectionInfoProvider(), sid, 65536, 65536)
+#endif
   {
   }
 

